app/addons/documents/actions.js

```diff
@@ -0,0 +1,164 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+define([
+  'app',
+  'api',
+  'addons/documents/resources',
+  'addons/documents/actiontypes'
+],
+function (app, FauxtonAPI, Documents, ActionTypes) {
+  var ActionHelpers = {
+    createNewDesignDoc: function (id, database) {
+      var designDoc = {
+        _id: id,
+        views: {
+        }
+      };
+
+      return new Documents.Doc(designDoc, {database: database});
+    },
+
+    findDesignDoc: function (designDocs, designDocId) {
+      return designDocs.find(function (doc) {
+        return doc.id === designDocId;
+      }).dDocModel();
+
+    }
+  };
+
+  return {
+    //helpers are added here for use in testing actions
+    helpers: ActionHelpers,
```

Comment:
If we want to differentiate the actions from the helpers, we could maybe just group the helpers in an object. Similarly, putting the actions into an actions property is tempting too, but getting verbose.

```javascript
return {
  helpers: {
    createNewDesignDoc: function() { ... },
    findDesignDoc: function() { ... },
  },
  toggleEditor: function () { ... } ,
  selectReduceChanged: function () { ... } 
  ...
}
```